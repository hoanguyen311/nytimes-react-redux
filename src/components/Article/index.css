.article {
  padding-bottom: 20px;
}

.article-header {
  margin-bottom: 20px;
  padding-top: 20px;
}

.article-headline {
  margin-bottom: 10px;
  font-size: 25px;
  line-height: 1;
}

.article-source {
  margin-bottom: 5px;
  line-height: 1;
  font-size: 16px;
  color: #818181;
}

.article-carousel {
  margin-bottom: 10px;
}

.article-image {
  max-width: 100%;
  display: block;
  margin: 0 auto;
}

.article-back-button {
  color: #999;
  font-size: 15px;
  margin-bottom: 10px;
  display: inline-block;
}
