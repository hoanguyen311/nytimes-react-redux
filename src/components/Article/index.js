import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { Skeleton, Icon } from 'antd';
import './index.css';

class Article extends Component {
  static propTypes = {
    id: PropTypes.string,
    item: PropTypes.shape({
      snippet: PropTypes.string,
      source: PropTypes.string,
      pub_date: PropTypes.string,
      headline: PropTypes.string,
      image: PropTypes.string,
      id: PropTypes.string,
      multimedia: PropTypes.arrayOf(PropTypes.shape({
        caption: PropTypes.string,
        url: PropTypes.string
      })),
    })
  };

  static defaultProps = {
    id: null,
    item: null
  };

  componentDidMount() {
    this.scrollToTop();
  }

  scrollToTop() {
    window && window.scroll && window.scroll(0, 0);
  }

  render() {
    const { item } = this.props;

    if (!item) {
      return (
        <Skeleton active loading paragraph={{ rows: 10 }} />
      )
    }

    const  { image, headline, snippet, formattedDate, source, web_url } = item;

    return (
      <article className="article">
        <header className="article-header">
          <Link className="article-back-button" to="/"><Icon type="arrow-left" theme="outlined" /> Back</Link>
          <h2 className="article-headline">{headline}</h2>
          {source && <h3 className="article-source">Source: {source}</h3>}
          {formattedDate && <small className="article-pub-date">{formattedDate}</small>}
        </header>
        <div className="article-carousel">
          {image && <img src={image} alt={headline} className="article-image" />}
        </div>
        <p className="article-content">{snippet}</p>
        <a target="_blank" rel="noopener noreferrer" href={web_url} title={headline}>Read more...</a>
      </article>
    )
  }
}

export default Article;
