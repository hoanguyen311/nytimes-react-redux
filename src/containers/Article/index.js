import { connect } from 'react-redux';
import Article from '@components/Article';

const mapStateToProps = (state, ownProps) => ({
  item: state.newsList.byId[ownProps.id]
});

export default connect(mapStateToProps)(Article);
