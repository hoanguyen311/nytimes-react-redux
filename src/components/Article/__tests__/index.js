import React from 'react';
import { StaticRouter } from 'react-router';
import renderer from 'react-test-renderer';
import Article from '../index';

it('render with default props', () => {
  const component = renderer.create(
    <Article />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('render with full props', () => {
  const props = {
    id: '123',
    item: {
      snippet: '123',
      source: 'abc',
      pub_date: 'ccc',
      headline: 'ccc',
      image: 'ccc',
      id: 'ccc'
    }
  };
  const component = renderer.create(<StaticRouter context={{}}><Article {...props} /></StaticRouter>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
