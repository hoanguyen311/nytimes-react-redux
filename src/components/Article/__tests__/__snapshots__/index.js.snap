// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with default props 1`] = `
<div
  className="ant-skeleton ant-skeleton-active"
>
  <div
    className="ant-skeleton-content"
  >
    <h3
      className="ant-skeleton-title"
      style={
        Object {
          "width": "38%",
        }
      }
    />
    <ul
      className="ant-skeleton-paragraph"
    >
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": "61%",
          }
        }
      />
    </ul>
  </div>
</div>
`;

exports[`render with full props 1`] = `
<article
  className="article"
>
  <header
    className="article-header"
  >
    <a
      className="article-back-button"
      href="/"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-arrow-left"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="arrow-left"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </i>
       Back
    </a>
    <h2
      className="article-headline"
    >
      ccc
    </h2>
    <h3
      className="article-source"
    >
      Source: 
      abc
    </h3>
  </header>
  <div
    className="article-carousel"
  >
    <img
      alt="ccc"
      className="article-image"
      src="ccc"
    />
  </div>
  <p
    className="article-content"
  >
    123
  </p>
  <a
    rel="noopener noreferrer"
    target="_blank"
    title="ccc"
  >
    Read more...
  </a>
</article>
`;
